internal/upstreamldap/upstreamldap.go

```diff
@@ -455,7 +464,7 @@ func (p *Provider) searchGroupsForUserDN(conn Conn, userDN string) ([]string, er
 		groupAttributeName = distinguishedNameAttributeName
 	}
 
-	var groups []string
+	groups := []string{}
 entries:
```

Comment:
You didn't change this line but I happen to notice that it is not needed anymore. It seems like there is no inner loop anymore for the continue statement to need this label.

Reply:
I tried taking it out and our integration tests failed 😬 
Weirdly, I didn't get any unit test failures, so that seems like a gap that I should try to fix before merging this.